include/linux/uaccess.h

```diff
@@ -74,18 +74,24 @@ static inline void force_uaccess_end(mm_segment_t oldfs)
 static __always_inline __must_check unsigned long
 __copy_from_user_inatomic(void *to, const void __user *from, unsigned long n)
 {
+	unsigned long res;
 	instrument_copy_from_user(to, from, n);
 	check_object_size(to, n, false);
-	return raw_copy_from_user(to, from, n);
+	res = raw_copy_from_user(to, from, n);
+	add_address(from, n, _RET_IP_, to);
```

Comment:
For copy_from_user() we wanted to only store the ranges that were actually read/written.
Usercopy functions return the number of bytes that was _not_ read or written, so you'll need to adjust the `len` parameter: `add_address(from, n - res, _RET_IP_, to);`